src/System.IO.Compression/tests/DeflateStreamAsyncTests.cs

```diff
@@ -0,0 +1,166 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Compression.Tests
+{
+    internal class ManualSyncMemoryStream : MemoryStream
+    {
+        public static async Task<ManualSyncMemoryStream> GetStreamFromFileAsync(String testFile, bool sync = false, bool strip = false)
+        {
+            var baseStream = await StreamHelpers.CreateTempCopyStream(testFile);
+            if (strip)
+            {
+                baseStream = StripHeaderAndFooter.Strip(baseStream);
+            }
+
+            var ms = new ManualSyncMemoryStream(sync);
+            await baseStream.CopyToAsync(ms);
+
+            ms.Position = 0;
+            return ms;
+        }
+
+        public ManualResetEventSlim manualResetEvent = new ManualResetEventSlim(false);
+
+        private bool isSync = false;
+        public ManualSyncMemoryStream(bool sync) : base()
+        {
+            isSync = sync;
+        }
+
+        public void SetManualEvent()
+        {
+            manualResetEvent.Set();
+            // Logger.LogInformation("Signal");
+        }
+
+        public void ResetManualEvent()
```

Comment:
Is this method needed?  It looks like it's only called in once place, and in that place the MRES is already not-signaled.